modules/api/src/main/java/org/apache/ignite/table/KeyValueView.java

```diff
@@ -491,4 +494,23 @@ public interface KeyValueView<K, V> extends DataStreamerTarget<Entry<K, V>> {
      * @see #getAndReplace(Transaction, Object, Object)
      */
     CompletableFuture<NullableValue<V>> getNullableAndReplaceAsync(@Nullable Transaction tx, K key, V val);
+
+    /**
+     * Criteria query over cache entries.
+     *
+     * @param tx Transaction or {@code null} to auto-commit.
+     * @param criteria If {@code null} then all entries will be returned.
+     */
+    default CriteriaQueryCursor<Entry<K, V>> criteriaQuery(@Nullable Transaction tx, @Nullable Criteria criteria) {
+        return criteriaQuery(tx, criteria, CriteriaQueryOptions.DEFAULT);
+    }
+
+    /**
+     * Criteria query over cache entries.
+     *
+     * @param tx Transaction or {@code null} to auto-commit.
+     * @param criteria If {@code null} then all entries will be returned.
+     * @param opts Criteria query options.
+     */
+    CriteriaQueryCursor<Entry<K, V>> criteriaQuery(@Nullable Transaction tx, @Nullable Criteria criteria, CriteriaQueryOptions opts);
```

Comment:
To avoid duplicating those declarations across multiple interfaces, let's move them to `CriteriaQuerySource` interface - similar to `DataStreamerTarget`.